MekHQ/src/mekhq/gui/enums/PersonnelTableModelColumn.java

```diff
@@ -871,14 +879,13 @@ public Comparator<?> getComparator(final Campaign campaign) {
             case RANK:
                 return new PersonRankStringSorter(campaign);
             case AGE:
-            case INJURIES:
-            case KILLS:
-            case XP:
-            case TOUGHNESS:
-            case EDGE:
-            case SPA_COUNT:
-            case IMPLANT_COUNT:
-                return new IntegerStringSorter();
+            case BIRTHDAY:
+            case RECRUITMENT_DATE:
+            case LAST_RANK_CHANGE_DATE:
+            case DUE_DATE:
+            case RETIREMENT_DATE:
+            case DEATH_DATE:
+                return new DateStringComparator();
```

Comment:
I'm amazed that there would be a date string comparator that can deal with different sorts of date formatting. Thinking of European DDMMYYYY and American MMDDYYYY here...
Since the cell renderer replaces the getValueAt() cell value with a display string anyway it would be possible to have getValueAt() return the date itself to have an easier time sorting it. Just a thought, no real reason to change anything here.

Reply:
MHQ has two date format options (short and long), and the display will always show it in the user's specified short format. That makes it rather easy to sort as we know the format it will use.

I don't want to make any changes to the table's data formats, so I'm going to leave it as a string instead of converting the setup to allow objects instead.